cni/log/logger.go

```diff
@@ -1,55 +1,43 @@
 package log
 
 import (
-	"context"
-	"fmt"
-	"os"
-
-	"go.uber.org/zap"
+	"github.com/Azure/azure-container-networking/log"
+	"github.com/Azure/azure-container-networking/zaplog"
 	"go.uber.org/zap/zapcore"
-	"gopkg.in/natefinch/lumberjack.v2"
 )
 
-type Config struct {
-	Level       zapcore.Level
-	LogPath     string
-	MaxSizeInMB int
-	MaxBackups  int
-	Name        string
-	Component   string
-}
-
-var Logger *zap.Logger
+const (
+	maxLogFileSizeInMb = 5
+	maxLogFileCount    = 8
+)
 
-// Initializes a Zap logger and returns a cleanup function so logger can be cleaned up from caller
-func Initialize(ctx context.Context, cfg *Config) {
-	Logger = newFileLogger(cfg)
+var (
+	LoggerVnetName      string
+	LoggerIpamName      string
+	LoggerTelemetryName string
+	LogLevel            int8
+)
 
-	go func() {
-		<-ctx.Done()
-		err := Logger.Sync()
-		if err != nil {
-			fmt.Println("failed to sync logger")
-		}
-	}()
+var LoggerVnetCfg = &zaplog.Config{
+	Level:       zapcore.DebugLevel,
+	LogPath:     log.LogPath + "azure-vnet.log",
+	MaxSizeInMB: maxLogFileSizeInMb,
+	MaxBackups:  maxLogFileCount,
+	Name:        LoggerVnetName,
 }
 
-func newFileLogger(cfg *Config) *zap.Logger {
-	logFileWriter := zapcore.AddSync(&lumberjack.Logger{
-		Filename:   cfg.LogPath,
-		MaxSize:    cfg.MaxSizeInMB,
-		MaxBackups: cfg.MaxBackups,
-	})
-
-	encoderConfig := zap.NewProductionEncoderConfig()
-	encoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
-	jsonEncoder := zapcore.NewJSONEncoder(encoderConfig)
-	logLevel := cfg.Level
-
-	core := zapcore.NewCore(jsonEncoder, logFileWriter, logLevel)
-	Logger = zap.New(core)
-	Logger = Logger.With(zap.Int("pid", os.Getpid()))
-	Logger = Logger.With(zap.String("component", cfg.Component))
+var LoggerIpamCfg = &zaplog.Config{
+	Level:       zapcore.DebugLevel,
+	LogPath:     log.LogPath + "azure-ipam.log",
+	MaxSizeInMB: maxLogFileSizeInMb,
+	MaxBackups:  maxLogFileCount,
+	Name:        LoggerIpamName,
+}
 
-	return Logger
+var LoggerTelemetryCfg = &zaplog.Config{
+	Level:       zapcore.Level(zapcore.Int8Type),
```

Comment:
Why the level is different for this one ?

Reply:
telemetry package was using "logLevel := acn.GetArg(acn.OptLogLevel).(zapcore.Level)"